database/api.go

```diff
@@ -244,15 +244,16 @@ func (d *api) DeleteWhere(obj record.Record, condition *qb.ConditionExpression)
 	})
 }
 
-func (d *api) enforceLimits(options *record.ListOptions) *record.ListOptions {
+func (d *api) enforceLimits(options qb.LimitOffset) qb.LimitOffset {
 	if options == nil {
-		options = record.NewListOptions(DefaultMaxLimit, 0)
+		options = qb.NewLimitOffset[uint]()
 	}
 	if d.configuration.MaxQueryLimit() != qb.NoLimit &&
-		options.Limit > d.configuration.MaxQueryLimit() {
+		options.Limit() > d.configuration.MaxQueryLimit() {
 		d.configuration.Logger().Warnf("limit %d exceeds max limit of %d", options.Limit,
 			d.configuration.MaxQueryLimit())
-		options.Limit = d.configuration.MaxQueryLimit()
+		options = qb.NewLimitOffset[uint]().SetOffset(options.Offset()).
```

Comment:
Why not just `SetLimit` on the existing options that we already prevent from being nil?

Reply:
LimitOffset does not have a Setter and we can't put one on without specifying the type.